bindings/python/test_x86.py

```diff
@@ -142,7 +239,14 @@ def print_string_hex(comment, str):
         for r in regs_write:
             print(" %s" %(insn.reg_name(r)), end="")
         print("")
-
+        
+    if insn.eflags:
+        updated_flags = []
+        for i in range(0,46):
```

Comment:
> because beyond 46, there is no flags anyway?

strictly speaking - "46" is not number of flag, but number of events that can occur with EFLAG register bits

according to [Intel 64 and IA-32...Manual:Appendix A: EFLAGS cross-reference](https://software.intel.com/en-us/articles/intel-sdm#combined):

Instructions could affect the flags in the EFLAGS register in the next **_6 ways_**:
1. Instruction tests flag (e.g.  X86_EFLAGS_**TEST**_NT)
2. Instruction modifies flag (either sets or resets depending on operands) (e.g. X86_EFLAGS_**MODIFY**_IF)
3. Instruction resets flag (e.g. X86_EFLAGS_**RESET**_PF)
4. Instruction sets flag (e.g. X86_EFLAGS_**SET**_CF)
5. Instruction's effect on flag is undefined (e.g. X86_EFLAGS_**UNDEFINED**_OF)
6. Instruction restores prior value of flag (e.g. X86_EFLAGS_**PRIOR**_CF)

Flags can be affected in next ways:
1. **OF** (Overflow flag)
MODIFY,RESET,TEST,PRIOR,UNDEFINED
In sum: **_5 events_**
2. **SF**(Sing flag)
MODIFY,RESET,TEST,PRIOR,UNDEFINED
In sum: **_5 events_**
3. **ZF**(Zero flag)
MODIFY,TEST,PRIOR,UNDEFINED
In sum: **_4 events_**
4. **AF**(Adjust flag)
MODIFY,RESET,TEST,PRIOR,UNDEFINED
In sum: **_5 events_**
5. **PF**(Parity flag)
MODIFY,TEST,PRIOR,UNDEFINED
In sum: **_4 events_**
6. **CF**(Carry flag)
MODIFY,RESET,SET,TEST,PRIOR,UNDEFINED
In sum: **_6 events_**
7. **TF**(Trap flag(single step))
MODIFY,RESET,PRIOR
In sum: **_3 events_**
8. **IF**(Interrupt enable flag)
MODIFY,RESET,SET,PRIOR
In sum: **_4 events_**
9. **DF**(Direction flag)
MODIFY,RESET,SET,TEST,PRIOR
In sum: **_5 events_**
10. **NT**(Nested task flag (286+ only), always 1 on 8086 and 186)
MODIFY,RESET,TEST,PRIOR
In sum: **_4 events_**
11. **RF**(Resume flag (386+ only))
MODIFY
In sum: **_1 events_**

==== All in all: 5+5+4+5+4+6+3+4+5+4+1 = **_46 events_** ====

IMHO, the origin and purpose of "46" is more obvious than "64", especially if someone explore code at first